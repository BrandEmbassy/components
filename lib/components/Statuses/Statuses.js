'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _bind = require('classnames/bind');

var _bind2 = _interopRequireDefault(_bind);

var _Statuses = require('./Statuses.css');

var _Statuses2 = _interopRequireDefault(_Statuses);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var cx = _bind2.default.bind(_Statuses2.default);

var Statuses = function (_Component) {
  _inherits(Statuses, _Component);

  function Statuses() {
    _classCallCheck(this, Statuses);

    return _possibleConstructorReturn(this, (Statuses.__proto__ || Object.getPrototypeOf(Statuses)).apply(this, arguments));
  }

  _createClass(Statuses, [{
    key: 'renderText',
    value: function renderText() {
      var status = this.props.status;

      //   // return <FormattedMessage id={`${status}`} defaultMessage="Default value if translation is not available" />

      if (!status) {
        return "empty";
      } else {
        return '' + status;
      };
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          styleName = _props.styleName,
          Reversed = _props.Reversed,
          status = _props.status;


      var className = cx(_Statuses2.default.Statuses, styleName, {
        Reversed: Reversed,
        New: status === "New",
        Open: status === "Open",
        Resolved: status === "Resolved",
        Pending: status === "Pending",
        Escalated: status === "Escalated",
        Closed: status === "Closed",
        Trashed: status === "Trashed"
      });

      return _react2.default.createElement(
        'div',
        { className: className },
        this.renderText()
      );
    }
  }], [{
    key: 'propTypes',
    get: function get() {
      return {
        text: _propTypes2.default.string,
        status: _propTypes2.default.string,
        styleName: _propTypes2.default.string
      };
    }
  }]);

  return Statuses;
}(_react.Component);

exports.default = Statuses;