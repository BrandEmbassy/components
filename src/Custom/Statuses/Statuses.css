.Statuses {
  position: relative;
  display: inline-flex;
  align-self: center;
  width: auto;
  height: auto;
  margin-right: 10px;
  padding: 3px 6px;
  line-height: inherit;
  font-size: 10px;
  font-weight: 700;
  color: #8d1616;
  border: solid 2px #dadada;
  border-radius: 3px;
  text-decoration: none;
  user-select: none;
  text-transform: uppercase;

  &.Orange {
    color: #F57D4A;
    border-color: #F57D4A;
  }

  &.Yellow {
    color: #F6AF4A;
    border-color: #F6AF4A;
  }
  
  &.Green {
    color: #31B272;
    border-color: #31B272;
  }

  &.Blue {
    color: #2A84F1;
    border-color: #2A84F1;
  }
  
  &.Purple {
    color: #9C418C;
    border-color: #9C418C;
  }
    
  &.Red {
    color: #E9575B;
    border-color: #E9575B;
  }
    
  &.Grey {
    color: #9A9A9A;
    border-color: #9A9A9A;
  }
}