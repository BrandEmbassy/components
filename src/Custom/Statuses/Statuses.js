import React, { Component } from 'react';
import styles from './Statuses.css';

export default class Statuses extends Component {
  render() {
    return (
      <div className={styles.Statuses}>
        Statuses
      </div>
    );
  }
}

