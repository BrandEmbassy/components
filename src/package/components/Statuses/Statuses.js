// @flow

import * as React from 'react'
// @flow-skip-next-line
import { renderToString } from 'react-dom/server'
import classNames from 'classnames/bind'
import styles from './Statuses.module.css'

const cx = classNames.bind(styles)

type Props = {
  text: ?(string | React.Node),
  status: string,
  styleName: ?string,
  Reversed: ?boolean,
  hasHover: ?boolean
};

export default function Statuses (props: Props) {
  const { styleName, Reversed, status, text, hasHover } = props

  const className = cx(
    styles.Statuses,
    styleName,
    status.charAt(0).toUpperCase() + status.slice(1).toLowerCase(),
    { Reversed, hasHover }
  )

  var altText = ''
  switch(typeof(text)) {
    case 'string':
      altText = text
      break;
    case React:Node:
      altText = renderToString(text ?? '')
      break;
  }
  const dataCy = `status-${status.toLowerCase() || altText.toLowerCase()}`

  return (
    <div data-cy={dataCy} className={className}>
      {text || status}
    </div>
  )
}
