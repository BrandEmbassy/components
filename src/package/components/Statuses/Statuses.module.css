.Statuses {
  position: relative;
  display: inline-flex;
  align-self: center;
  width: auto;
  height: auto;
  margin-right: 6px;
  padding: 3px 6px;
  line-height: 10px;
  font-size: 10px;
  font-weight: 700;
  color: #8d1616;
  border: solid 2px #dadada;
  border-radius: 3px;
  text-decoration: none;
  user-select: none;
  text-transform: uppercase;
}
.Statuses.hasHover {
  cursor: pointer;
}
.Statuses.hasHover:hover {
  filter: url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg"><filter id="filter"><feComponentTransfer color-interpolation-filters="sRGB"><feFuncA type="table" tableValues="0 0.8" /></feComponentTransfer></filter></svg>#filter');
  filter: opacity(0.8);
  transition: 0.18s;
}
.Statuses.New {
  color: #f57d4a;
  border-color: #f57d4a;
}
.Statuses.New.Reversed {
  color: #ffffff;
  background-color: #f57d4a;
}
.Statuses.Open {
  color: #f6af4a;
  border-color: #f6af4a;
}
.Statuses.Open.Reversed {
  color: #ffffff;
  background-color: #f6af4a;
}
.Statuses.Resolved {
  color: #31b272;
  border-color: #31b272;
}
.Statuses.Resolved.Reversed {
  color: #ffffff;
  background-color: #31b272;
}
.Statuses.Pending {
  color: #2a84f1;
  border-color: #2a84f1;
}
.Statuses.Pending.Reversed {
  color: #ffffff;
  background-color: #2a84f1;
}
.Statuses.Escalated {
  color: #9c418c;
  border-color: #9c418c;
}
.Statuses.Escalated.Reversed {
  color: #ffffff;
  background-color: #9c418c;
}
.Statuses.Closed {
  color: #e9575b;
  border-color: #e9575b;
}
.Statuses.Closed.Reversed {
  color: #ffffff;
  background-color: #e9575b;
}
.Statuses.Trashed {
  color: #9a9a9a;
  border-color: #9a9a9a;
}
.Statuses.Trashed.Reversed {
  color: #ffffff;
  background-color: #9a9a9a;
}
.Statuses.Checked {
  color: #9a9a9a;
  border-color: #9a9a9a;
}
.Statuses.Checked.Reversed {
  color: #ffffff;
  background-color: #9a9a9a;
}
.Statuses.Approval {
  border-color: #ca4b5b;
  color: #ca4b5b;
}
.Statuses.Approval.Reversed {
  background-color: #ca4b5b;
  color: #f7bac2;
}
.Statuses.Assigntoyou {
  border-color: #ca4b5b;
  color: #ca4b5b;
}
.Statuses.Assigntoyou.Reversed {
  background-color: #ca4b5b;
  color: #f7bac2;
}
