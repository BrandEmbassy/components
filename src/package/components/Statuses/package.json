{
  "private": true,
  "name": "Statuses",
  "main": "./Statuses.js"
}
