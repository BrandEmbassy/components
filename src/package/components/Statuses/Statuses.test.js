import React from 'react'
import Statuses from './Statuses'

describe('Statuses', () => {
  const dataProvider = [
    { text: 'custom label', status: 'new' },
    { status: 'new' }
  ]

  dataProvider.forEach(dataSet => {
    it(`should render status with ${JSON.stringify(dataSet)}`, () => {
      const wrapper = shallow(<Statuses {...dataSet} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
