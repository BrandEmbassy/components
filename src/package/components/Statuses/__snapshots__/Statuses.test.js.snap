// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statuses should render status with {"status":"resolved"} 1`] = `
<div
  className="Resolved"
  data-cy="status-resolved"
>
  resolved
</div>
`;

exports[`Statuses should render status with {"text":"custom label","status":"new","hasHover":true} 1`] = `
<div
  className="New hasHover"
  data-cy="status-new"
>
  custom label
</div>
`;
